src/pkg/cli/loader.go

```diff
@@ -75,16 +58,29 @@ func loadCompose(filePath string, projectName string, overrideProjectName bool)
 	}
 
 	// Hack: Fill in the missing environment variables that were stripped by the normalization process
+	skipNormalizationOpts := append(loadOpts, func(o *loader.Options) {
+		o.SkipNormalization = true // Normalization strips environment variables keys that does not have an value
+	})
+
+	// Disable logrus output to prevent double warnings from compose-go
+	currentOutput := logrus.StandardLogger().Out
+	logrus.SetOutput(io.Discard)
+	rawProj, err := loader.Load(loadCfg, skipNormalizationOpts...)
+	logrus.SetOutput(currentOutput)
+	if err != nil {
+		return nil, err // there's no good reason this should fail, since we've already loaded the project
+	}
+
 	// TODO: file a PR to compose-go to add option to keep unset environment variables
-	for i, service := range rawProj.Services {
-		for key, value := range service.Environment {
-			svc := project.Services[i]
-			if svc.Environment[key] == nil {
-				if svc.Environment == nil {
-					svc.Environment = make(map[string]*string)
+	for name, rawService := range rawProj.Services {
+		for key, value := range rawService.Environment {
+			service := project.Services[name]
+			if service.Environment[key] == nil {
+				if service.Environment == nil {
```

Comment:
service.Environment is de-referenced above before checking that it's not nil.

Reply:
Strangely enough, that's valid because it's a `map`: you can deref a `nil` map (but can't write to it)